parser_library/src/processing/preprocessors/cics_preprocessor.cpp

```diff
@@ -1316,14 +1313,37 @@ class cics_preprocessor final : public preprocessor
             skip_continuation = is_continued(text);
         }
 
-        try_general_injections();
+        do_general_injections();
         if (!std::exchange(m_end_seen, true) && !asm_xopts_allowed) // actual code encountered
             inject_no_end_warning();
 
         return document(std::move(m_result));
     }
 
     cics_preprocessor_options current_options() const { return m_options; }
+
+    void do_highlighting(const semantics::preprocessor_statement_si& stmt,
+        semantics::source_info_processor& src_proc,
+        size_t continue_column = 15) const override
+    {
+        preprocessor::do_highlighting(stmt, src_proc, continue_column);
+
+        size_t lineno = stmt.m_details.stmt_r.start.line;
+        for (size_t i = 0; i < m_logical_line.segments.size(); ++i)
+        {
+            const auto& segment = m_logical_line.segments[i];
+
+            if (!segment.continuation.empty())
+                m_src_proc.add_hl_symbol(token_info(
+                    range(position(lineno + i, 71), position(lineno + i, 72)), semantics::hl_scopes::continuation));
```

Comment:
please check if there is something like the DBCS mode in the DB2 preprocessor

Reply:
There is -> `TODO` added